### PR TITLE
feat: add dropdown to select token on swap

### LINKED ISSUE #131: On swap when inverting values its showing insufficient liquidity

### LINKED ISSUE #174: Enable token select dropdown on swap
On Swap the user should be able to select on the dropdown the token he, the idea is to start with no token.

- [ ] One side should be always ETH and not be able to select. When clicking to invert this rule also inverts.
- [ ] This screen should only show `DAI` for now.

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #174
✨ Fix #131 

Kapture 2022-05-20 at 21 02 46

### CODE REVIEW

**Comment by pedronauck** [comment]
got #131 fixed as well :v:

Kapture 2022-05-20 at 21 10 45

File: packages/app/src/pages/SwapPage/SwapComponent.tsx [hunk 1]
```diff
@@ -40,20 +40,22 @@ export function SwapComponent({
 
   const handleInvertCoins = () => {
     if (activeInput.current === ActiveInput.to) {
-      activeInput.current = ActiveInput.from;
+      const from = fromInput.amount;
       fromInput.setAmount(toInput.amount);
-      toInput.setAmount(null);
+      toInput.setAmount(from);
     } else {
-      activeInput.current = ActiveInput.to;
```
Comment (luizstacio) [inline packages/app/src/pages/SwapPage/SwapComponent.tsx:47]: This is causing the issue you see when inverting fields

Comment (luizstacio) [reply packages/app/src/pages/SwapPage/SwapComponent.tsx:47]: We use reference here because we need a nonstate delayed change.

Comment (luizstacio) [reply packages/app/src/pages/SwapPage/SwapComponent.tsx:47]: https://user-images.githubusercontent.com/3941923/169627280-200c409c-6db0-4329-8632-cc8b4b5aa736.mov

File: packages/app/src/pages/SwapPage/SwapComponent.tsx [hunk 2]
```diff
@@ -122,7 +128,7 @@ export function SwapComponent({
           {...fromInput.getInputProps()}
           {...(activeInput.current === ActiveInput.to && { isLoading })}
           autoFocus={activeInput.current === ActiveInput.from}
-          coinSelectorDisabled={true}
+          coinSelectorDisabled={coinFrom?.symbol === "ETH"}
```
Comment (luizstacio) [inline packages/app/src/pages/SwapPage/SwapComponent.tsx:131]: You can use CoinETH const form `~/lib/constants` to check here.

File: packages/app/src/components/CoinsListDialog.tsx [hunk 3]
```diff
@@ -13,6 +13,11 @@ const style = {
   noResults: `px-6 py-4 border-t border-gray-700`,
 };
 
+function filterByValid(coin: Coin) {
+  // this is just for now until we have tokens coming from backend
```
Comment (luizstacio) [inline packages/app/src/components/CoinsListDialog.tsx:17]: Just add a TODO: for tracking

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/pages/SwapPage/SwapComponent.tsx (hunk 2 above)
Comment (pedronauck) [reply packages/app/src/pages/SwapPage/SwapComponent.tsx]: ✅

File: packages/app/src/components/CoinsListDialog.tsx (hunk 3 above)
Comment (pedronauck) [reply packages/app/src/components/CoinsListDialog.tsx:18]: ✅

File: packages/app/src/pages/SwapPage/SwapComponent.tsx (hunk 1 above)
Comment (luizstacio) [reply packages/app/src/pages/SwapPage/SwapComponent.tsx:47]: Thanks for adding the `startTransition`

**Pull request merged**